common/views/components/ImageViewer/ImageViewer.js

```diff
@@ -1,60 +1,148 @@
 // @flow
 import { useState, useEffect } from 'react';
+import fetch from 'isomorphic-unfetch';
 import IIIFResponsiveImage from '../IIIFResponsiveImage/IIIFResponsiveImage';
 import Control from '../Buttons/Control/Control';
 import { spacing, classNames } from '../../../utils/classnames';
-import { trackEvent } from '../../../utils/ga';
-import dynamic from 'next/dynamic';
+import Raven from 'raven-js';
 
-const ImageViewerImage = dynamic(import('./ImageViewerImage'), { ssr: false });
+function getTileSources(data) {
+  return [
+    {
+      '@context': 'http://iiif.io/api/image/2/context.json',
+      '@id': data['@id'],
+      height: data.height,
+      width: data.width,
+      profile: ['http://iiif.io/api/image/2/level2.json'],
+      protocol: 'http://iiif.io/api/image',
+      tiles: [
+        {
+          scaleFactors: [1, 2, 4, 8, 16, 32],
+          width: 400,
+        },
+      ],
+    },
+  ];
+}
 
-type ViewerContentProps = {|
+type ImageViewerProps = {|
   id: string,
-  classes: string,
-  viewerVisible: boolean,
+  src: string,
+  srcSet: string,
+  width: number,
+  height?: number,
+  canvasOcr: ?string,
   infoUrl: string,
+  lang: ?string,
 |};
 
-const ViewerContent = ({
+const ImageViewer = ({
   id,
-  classes,
-  viewerVisible,
+  width,
+  height,
+  canvasOcr,
+  lang,
   infoUrl,
-}: ViewerContentProps) => {
-  const handleRotate = event => {
-    trackEvent({
-      category: 'Control',
-      action: 'rotate ImageViewer',
-      label: id,
-    });
-  };
-
-  const handleZoomIn = event => {
-    trackEvent({
-      category: 'Control',
-      action: 'zoom in ImageViewer',
-      label: id,
-    });
-  };
-
-  const handleZoomOut = event => {
-    trackEvent({
-      category: 'Control',
-      action: 'zoom out ImageViewer',
-      label: id,
-    });
-  };
+  src,
+  srcSet,
+}: ImageViewerProps) => {
+  const [viewer, setViewer] = useState(null);
+  const zoomStep = 0.5;
+
+  useEffect(() => {
+    if (viewer) {
+      fetch(infoUrl)
+        .then(response => response.json())
+        .then(data => {
+          viewer.open(getTileSources(data));
+        });
+    }
```

Comment:
If we don't have the viewer?

Reply:
We do nothing and load a static image instead